test/bats_pipe.bats

```diff
@@ -0,0 +1,841 @@
+
+  [ "$status" -eq 42 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" = 'Got 0 args.' ]
+  [ "${lines[1]}" = 'stdin:' ]
+  [ "${lines[2]}" = $'\tWill return status 42.' ]
+}
+
+@test "run bats_pipe for 1st error status and fail on third of three" {
+  run bats_pipe -1 returns_with_given_code 0 \| consume_stdin_and_returns_with_given_code 0 \| consume_stdin_and_returns_with_given_code 42
+
+  [ "$status" -eq 0 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 1 ]
+  [ "${lines[0]}" = 'Will return status 42.' ]
+}
+
+@test "run bats_pipe for 1st error status and fail on all of three" {
+  run bats_pipe -1 returns_with_given_code 42 \| consume_stdin_and_returns_with_given_code 84 \| consume_stdin_and_returns_with_given_code 4
+
+  [ "$status" -eq 84 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 1 ]
+  [ "${lines[0]}" = 'Will return status 4.' ]
+}
+
+@test "run bats_pipe for 2nd error status and fail on first of three" {
+  run bats_pipe -2 returns_with_given_code 42 \| describe_input \| describe_input
+
+  [ "$status" -eq 0 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 5 ]
+  [ "${lines[0]}" = 'Got 0 args.' ]
+  [ "${lines[1]}" = 'stdin:' ]
+  [ "${lines[2]}" = $'\tGot 0 args.' ]
+  [ "${lines[3]}" = $'\tstdin:' ]
+  [ "${lines[4]}" = $'\t\tWill return status 42.' ]
+}
+
+@test "run bats_pipe for 2nd error status and fail on first of three with arg separator" {
+  run bats_pipe -2 -- returns_with_given_code 42 \| describe_input \| describe_input
+
+  [ "$status" -eq 0 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 5 ]
+  [ "${lines[0]}" = 'Got 0 args.' ]
+  [ "${lines[1]}" = 'stdin:' ]
+  [ "${lines[2]}" = $'\tGot 0 args.' ]
+  [ "${lines[3]}" = $'\tstdin:' ]
+  [ "${lines[4]}" = $'\t\tWill return status 42.' ]
+}
+
+@test "run bats_pipe for 2nd error status and fail on second of three" {
+  run bats_pipe -2 returns_with_given_code 0 \| consume_stdin_and_returns_with_given_code 42 \| describe_input
+
+  [ "$status" -eq 0 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" = 'Got 0 args.' ]
+  [ "${lines[1]}" = 'stdin:' ]
+  [ "${lines[2]}" = $'\tWill return status 42.' ]
+}
+
+@test "run bats_pipe for 2nd error status and fail on third of three" {
+  run bats_pipe -2 returns_with_given_code 0 \| consume_stdin_and_returns_with_given_code 0 \| consume_stdin_and_returns_with_given_code 42
+
+  [ "$status" -eq 42 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 1 ]
+  [ "${lines[0]}" = 'Will return status 42.' ]
+}
+
+@test "run bats_pipe for 2nd error status and fail on all of three" {
+  run bats_pipe -2 returns_with_given_code 42 \| consume_stdin_and_returns_with_given_code 84 \| consume_stdin_and_returns_with_given_code 4
+
+  [ "$status" -eq 4 ]
+  echo "$output" >&2
+  [ "${#lines[@]}" -eq 1 ]
+  [ "${lines[0]}" = 'Will return status 4.' ]
+}
+
+@test "run bats_pipe for Nth error status too large" {
```

Comment:
If we added support for negative indices above, please add tests for the range check and actual return code position here.

Reply:
Added tests for `--returned-status` as discussed in other comment thread.